src/goCheck.ts

```diff
@@ -106,6 +106,43 @@ function runTool(args: string[], cwd: string, severity: string, useStdErr: boole
 	});
 }
 
+/**
+ * Check if the given path is a directory.
+ * @param filePath Path to the potential directory
+ */
+function isDirectory(filePath: string): boolean {
+	try {
+		return fs.statSync(filePath).isDirectory();
+	} catch (e) {
+		return false;
+	}
+}
+
+/**
+ * List directories (including the given dirPath) which contain files with the '_test.go' suffix.
+ * This ignores hidden directories and their sub-directories.
+ * @param dirPath Directory in which to start searching
+ */
+function findTestDirectories(dirPath: string): string[] {
```

Comment:
I appreciate the work here. Consider this
- User with a very large workspace with deeply nested directories
- User with auto-save on

The directories check here is synchronous.
We later wait for all promises (build, lint, vet) to finish before showing diagnostics to the user. 

So you can imagine for each save, this might delay results more than needed.
Note: We have #1015 tracking to find a way to optimize this.

Also, I'd much rather prefer a better way to compile test files from Go itself.

Until then, we can skip trying to compile the test files in the whole workspace.